DeliveryApi/ApiHandlers/src/com/ilmlf/delivery/api/handlers/PopulateFarmDb.java

```diff
@@ -96,78 +124,150 @@
-        String line;
+    String returnJson = 
+        this.buildReturnJson(customResourceEvent.getRequestType(), 
+            customResourceEvent.getPhysicalResourceId(), runScript);
+    return returnJson;
+  }
+
+  /**
+   * Extracts the list of SQL statements from a file.
+   *
+   * @param fileName of the SQL file
+   */
+  public List<String> extractSqlStatementsFromFile(String fileName) {
+    try (FileReader fr = new FileReader(fileName);
+        BufferedReader br = new BufferedReader(fr)) {
+
+      return this.extractSqlStatements(br);
+      
+    } catch (IOException e) {
+      logger.error(e.getMessage(), e);
+      throw new RuntimeException(e);
+    }
+  }
   
-        while ((line = br.readLine()) != null) {
-          if (line.contains("{{username}}") || line.contains("{{password}}")) {
-            line = replaceUserDbProxyCredentials(line);
-          }
+  /**
+   * Extracts the SQL statements from the BufferedReader into an array list.
+   *
+   * @param br the Buffered Reader
+   * @return the list of SQL statements as an ArrayList String
+   * @throws IOException on reading error
+   */
+  public List<String> extractSqlStatements(BufferedReader br) throws IOException {
+    List<String> sqlStmts = new ArrayList<String>();
+    StringBuilder sb = new StringBuilder();
+    String line;
+
+    while ((line = br.readLine()) != null) {
+      sb.append(line);
+      if (line.contains(";")) {
+        String query = sb.toString();
+        sqlStmts.add(query);
+        sb = new StringBuilder(); // reinitialize, otherwise will keep appending
+      }
+    }
+    return sqlStmts;
+  }
   
-          sb.append(line);
-          if (line.contains(";")) {
-            String query = sb.toString();
-            stmt.execute(query);
-            sb = new StringBuilder(); // reinitialize, otherwise will keep appending
-          }
-        }
+  /**
+   * Executes the list of SQL statements.
+   *
+   * @param sqlStmts is the list of sql statements to be executed
+   */
+  public void executeSqlStatements(List<String> sqlStmts) {
+    try (Statement stmt = con.createStatement()) {
+      for (String sqlStmt : sqlStmts) {
+        stmt.execute(sqlStmt);
+      }
+    } catch (SQLException e) {
+      logger.error(e.getMessage(), e);
+      throw new RuntimeException(e);
+    } 
+  }
   
-      // Rethrow checked exceptions as a runtime exception as
-      // the overridden method signature doesn't throw any exceptions
-      } catch (SQLException e) {
-        logger.error(e.getMessage(), e);
-        throw new RuntimeException(e);
-      } catch (IOException e) {
-        logger.error(e.getMessage(), e);
-        throw new RuntimeException(e);
+  /**
+   * Should the SQL script be executed or not. 
+   *
+   * @param requestType from the Custom resource
+   * @return true if the request type is Update or Create
+   */
+  public boolean isExecuteScript(String requestType) {
+    if (null != requestType) {
```

Comment:
a bit weird to execute the script if you don't know the requestType ? 
But if that is really needed, wouldn't it be easier to just write 
`if(REQUEST_DELETE.equals(requestType)) return false;` ?

Reply:
requesttype == null when you test the lambda on the console. 
Regarding your if-not-delete suggestion, it is logically the same (and shorter).. but then people might not infer the requesttype==null case. I'll change it but add an explanation about it in the javadoc.